### ISSUE TITLE
Compress middleware excludedContentTypes not working as expected

### ISSUE THREAD

**Issue by ddtmachado** [traefik/traefik#6810]
### Do you want to request a *feature* or report a *bug*?

Bug

### What did you do?

Setup `Compress` middleware with `excludedContentTypes` = `text/plain`

### What did you expect to see?

A response with content type `text/plain` not being compressed.

### What did you see instead?

```
$ curl -v -H "Accept-Encoding: deflate,gzip" -H "Accept: plain/text" localhost/whoami/data?size=5000
> GET /whoami/data?size=5000 HTTP/1.1
> Host: localhost
> User-Agent: curl/7.64.0
> Accept-Encoding: deflate,gzip
> Accept: plain/text
> 
< HTTP/1.1 200 OK
< Content-Encoding: gzip
< Content-Type: text/plain; charset=utf-8
< Date: Thu, 14 May 2020 18:39:00 GMT
< Vary: Accept-Encoding
< Content-Length: 80
< 
```

### Output of `traefik version`: (_What version of Traefik are you using?_)

```
Version:      2.2.1
Codename:     chevrotin
Go version:   go1.14.2
Built:        2020-04-29T18:02:09Z
OS/Arch:      linux/amd64
```

### What is your environment & configuration (arguments, toml, provider, platform, ...)?

```bash
docker service create --name whoami \
--network traefikee_ingress \
-l traefik.enable=true \
-l traefik.http.middlewares.who-compress.compress=true \
-l "traefik.http.middlewares.who-compress.compress.excludedcontenttypes=text/plain" \
-l traefik.http.routers.who.entrypoints=web \
-l "traefik.http.routers.who.rule=PathPrefix(\`/whoami\`)" \
-l traefik.http.services.who.loadbalancer.server.port=80 \
-l "traefik.http.routers.who.middlewares=who-strip,who-compress" \
-l "traefik.http.middlewares.who-strip.stripprefix.prefixes=/whoami" \
containous/whoami
```

**Comment by xdxiaodong** [comment]
+1

**Comment by traefiker** [comment]
Closed by #7888.
